### ISSUE TITLE
Can we deal with user code (packages) causing kernel failures

### ISSUE THREAD

**Issue by DonJayamanne** [microsoft/vscode-jupyter#9375]
* We know tensorflow incorrectly installed could cause the kernel to crash
Such as importing tensorflow See https://github.com/microsoft/vscode-jupyter/issues/7600 , https://github.com/microsoft/vscode-jupyter/issues/4645
* Similarly other packages
	* https://github.com/microsoft/vscode-jupyter/issues/9368

Misc:
* https://github.com/microsoft/vscode-jupyter/issues/9283

**Suggestions**
* Improve error message dispalyed in the popup
	* When kernel crashes provide a more meaningful message about what happened, users might not know what a kernel is, hence might ignore the error message or wouldn't be able to connect execution of cell to an error message about `kernel failure`
* Include error message in cell output
	* This way they know what cell caused the failure
* Should we pay specific focus to packages such as tensorflow, numpy, fastparquet (and provide a link to a Wiki)
	* Note: We do this today for kernel startup failures (e.g. if we have a DLL exception, we provide error information and a link to a WIKI)	    
	* Provide instructions in the error output specific to some packages (with links to WIKI)

**Comment by DonJayamanne** [comment]
Run a cell with the following code (cause the kernel to crash)
```python
import IPython
app = IPython.Application.instance()
app.kernel.do_shutdown(True)
```

Output should contrain a link to wiki as follows:
<img width="831" alt="Screen Shot 2022-05-04 at 06 02 31" src="https://user-images.githubusercontent.com/1948812/166556788-6783cbc9-19e3-4f00-8827-1b511b241dad.png">

**Comment by rchiodo** [comment]
So this is the same fix as #9588.

/verified
